s3_storage_client_test.go

```diff
@@ -285,6 +285,22 @@ func TestGetHeaderUnexpectedError(t *testing.T) {
 	}
 }
 
+func TestGetHeaderNonApiError(t *testing.T) {
+	othErr := errors.New("something went wrong here")
+	meta := fileMetadata{
+		client:    s3.New(s3.Options{}),
+		stageInfo: &execResponseStageInfo{Location: ""},
+		mockHeader: mockHeaderAPI(func(ctx context.Context, params *s3.HeadObjectInput, optFns ...func(*s3.Options)) (*s3.HeadObjectOutput, error) {
+			return nil, othErr
+		}),
+	}
+
+	header, err := new(snowflakeS3Client).getFileHeader(&meta, "file.txt")
+	assertEqualF(t, header, nil, fmt.Sprintf("expected nil header, got: %v", header))
```

Comment:
We have `assertNilF(E)` to check nullability.

Reply:
got it; used assertNilE and also added assertNotNilE (was only F available)